Missionframework/scripts/server/ai/AA_turret_manager.sqf

```diff
@@ -0,0 +1,88 @@
+waitUntil {!isNil "save_is_loaded"};
+waitUntil {!isNil "GRLIB_vehicle_to_military_base_links"};
+waitUntil {!isNil "blufor_sectors"};
+waitUntil {save_is_loaded};
+
+if (GRLIB_difficulty_modifier == 0) exitWith {};											// no AA turrets on easiest difficulty level
+if (isNil "KPLIB_o_AA_Turrets") exitWith {};													// leave, if there are no AA turrets defined in currently played preset
+private _AA_Killed_Turrets = 0;																// counter of killed AA turrets
+if (isNil "AA_used_positions") then {AA_used_positions = []};								// define array containing all currently used positions
+if (isNil "AA_backland_turrets") then {AA_backland_turrets = []};							// define array containing all turrets corresponding to a used position
+
+while {GRLIB_endgame == 0} do {
+    private _sleeptime =  (1800 + (random 1800)) / (([] call KPLIB_fnc_getOpforFactor) * GRLIB_csat_aggressivity);
```

Comment:
Aha. This is why the slower declaration of private variables is done outside loops. I already guessed so, but I was only able to find information, that 

private _varname = "this is my new variable"; // since Arma 3 v1.53 only

// identical, but less performant
private "_varname";

Well, now that you told me, I know, that this is only valid outside loops. They really should allow qualified people to further comment on their Scripting Commands.

And I am not discouraged by your extended reply. I am honoured, that you took your time to correct me and give such an extensive feedback. It would be discouraging, if some PR was closed without any info, why it was closed.

Reply:
Concerning the test. I host a derivate of v0.96.7a, then I play as commander. It's true, no other players are on the server, but for me, I reached the goal. I am no longer able to fly high above the enemy territory without fear of being shot down at any time.
So, it works. I also checked the spawned turrets, and all seems well. 
Due to lack of multiplayer, I don't know, if any other player would be attacked too by those turrets. But I noobishly guess, that this should be the case. I should maybe host a server and invite some people testing.

Reply:
It's about the environment, not the player count. To ensure locality stuff was handled correctly and there is no issue when the player isn't also the server.

Reply:
"...and going away from these old and ugly endless loops...."
Yes, I try to avoid them, but due to lack of finding and defining a proper 'event', I had to stick with the endless loop. If you would be so kind and push me in the right direction? Best would be a little example. I will see, if and how I manage to turn the loop into an event driven thing.

Reply:
Please keep your comments to the actual topic of the thread. In the review comment for that line the topic isn't the environment and also not about event handling instead of loops, etc.

Reply:
Speaking of event driven. What is your opinion on running the AA manager only when KPLIB_enemyReadiness changes? I am not aware, on how doing that with an event. I would have to do some loop again for watching KPLIB_enemyReadiness, right?